kanidmd/src/lib/core/https/mod.rs

```diff
@@ -1226,16 +466,35 @@ pub fn create_https_server(
             })?;
     };
 
-    tserver.at("/status").get(self::status);
+    // ==== Some routes can be cached - these are here:
+    let mut tserver_cacheable = tserver.at("");
+    tserver_cacheable.with(CacheableMiddleware::default());
+
+    let mut account_route_cacheable = tserver_cacheable.at("/v1/account");
 
-    let mut well_known = tserver.at("/well-known");
+    // We allow caching of the radius token.
+    account_route_cacheable
+        .at("/:id/_radius/_token")
+        .get(account_get_id_radius_token);
+    // We allow clients to cache the unix token.
+    account_route_cacheable
+        .at("/:id/_unix/_token")
+        .get(account_get_id_unix_token);
+
+    // ==== These routes can not be cached
+
+    let mut appserver = tserver.at("");
+    appserver.with(NoCacheMiddleware::default());
```

Comment:
Isn't there still a `CacheableMiddleware` still in `tserver.at("")`? Or does this somehow only apply to the `tserver_cacheable` reference? I'm confused how these are are enforced to be disjoint.

Reply:
So what happens is this is building a tree basicly. When we have the appserver from at("") this means "apply this middle ware at this route point" and then we build a set of subroutes that don't get cached through the appserver builder. The same way that later we use a different middleware that does apply the cache headers.

Reply:
Ah, so when we insert the `NoCacheMiddleware`, we're making a new branch from `tserver.at("")` that we can add route functionality on. Whereas when we insert `CacheableMiddleware` above, that's it's own branch off of `tserver.at("")` that doesn't affect the route at `""` directly. Since the two middlewares are on their own branches of the tree, they're disjoint?

Reply:
Yep, exactly. So all the routes are still correct but this just lets us apply different middlewares.